in/command.go

```diff
@@ -80,59 +80,60 @@ func (command *Command) Run(destinationDir string, request Request) (Response, e
 		isInitialVersion = request.Source.InitialVersion != "" && request.Version.VersionID == request.Source.InitialVersion
 	}
 
-	if isInitialVersion {
-		if request.Source.InitialContentText != "" || request.Source.InitialContentBinary == "" {
-			err = command.createInitialFile(destinationDir, path.Base(remotePath), []byte(request.Source.InitialContentText))
-			if err != nil {
-				return Response{}, err
+	if !request.Source.SkipDownload {
+		if isInitialVersion {
+			if request.Source.InitialContentText != "" || request.Source.InitialContentBinary == "" {
+				err = command.createInitialFile(destinationDir, path.Base(remotePath), []byte(request.Source.InitialContentText))
+				if err != nil {
+					return Response{}, err
+				}
 			}
-		}
-		if request.Source.InitialContentBinary != "" {
-			b, err := base64.StdEncoding.DecodeString(request.Source.InitialContentBinary)
-			if err != nil {
-				return Response{}, errors.New("failed to decode initial_content_binary, make sure it's base64 encoded")
+			if request.Source.InitialContentBinary != "" {
+				b, err := base64.StdEncoding.DecodeString(request.Source.InitialContentBinary)
+				if err != nil {
+					return Response{}, errors.New("failed to decode initial_content_binary, make sure it's base64 encoded")
+				}
+				err = command.createInitialFile(destinationDir, path.Base(remotePath), b)
+				if err != nil {
+					return Response{}, err
+				}
 			}
-			err = command.createInitialFile(destinationDir, path.Base(remotePath), b)
+		} else {
+			err = command.downloadFile(
+				request.Source.Bucket,
+				remotePath,
+				versionID,
+				destinationDir,
+				path.Base(remotePath),
+			)
 			if err != nil {
 				return Response{}, err
 			}
-		}
-	} else {
-		err = command.downloadFile(
-			request.Source.Bucket,
-			remotePath,
-			versionID,
-			destinationDir,
-			path.Base(remotePath),
-		)
-		if err != nil {
-			return Response{}, err
-		}
 
-		if request.Params.Unpack {
-			destinationPath := filepath.Join(destinationDir, path.Base(remotePath))
-			mime := archiveMimetype(destinationPath)
-			if mime == "" {
-				return Response{}, fmt.Errorf("not an archive: %s", destinationPath)
+			if request.Params.Unpack {
+				destinationPath := filepath.Join(destinationDir, path.Base(remotePath))
+				mime := archiveMimetype(destinationPath)
+				if mime == "" {
+					return Response{}, fmt.Errorf("not an archive: %s", destinationPath)
+				}
+
+				err = extractArchive(mime, destinationPath)
+				if err != nil {
+					return Response{}, err
+				}
 			}
-
-			err = extractArchive(mime, destinationPath)
-			if err != nil {
+			url = command.urlProvider.GetURL(request, remotePath)
+			if err = command.writeURLFile(destinationDir, url); err != nil {
```

Comment:
I don't think this should be skipped, either - they still may want the URL to the file even though they didn't download it.